### PR TITLE
rust nasl syntax tokenizer

### PR DESCRIPTION
**Pull request by nichtsfrei**
Change: add tokenizer for single character tokens

On top of Cursor a Tokenizer is needed to tokenize the given source
code. This initial Tokenizer can handle NASL single character tokens.

A Tokenizer implements the Iterator interface and can be used as that:

```
let single_char_example = "() [ ] { } , . - + % ; / * : ~ & | ^";
let actual: Vec<Token> = Tokenizer::new(single_char_example).collect();
```

Depends on: https://github.com/greenbone/openvas-scanner/pull/1222

### CODE REVIEW

File: rust/nasl-syntax/src/token.rs [hunk 1]
```diff
@@ -0,0 +1,773 @@
...
+            }
+        };
+        if let Some(base) = may_base {
+            self.cursor.skip_while(base.verifier());
+            // we only allow float numbers in base10
+            if base == Base10 && self.cursor.peek(0) == '.' && self.cursor.peek(1).is_numeric() {
+                self.cursor.advance();
+                self.cursor.skip_while(base.verifier());
+            }
+            if start == self.cursor.len_consumed() {
```
Comment (nichtsfrei) [inline rust/nasl-syntax/src/token.rs:359]: put a comment here

File: rust/nasl-syntax/src/token.rs [hunk 2]
```diff
@@ -0,0 +1,773 @@
...
+        match self.cursor.peek(0) {
+            '=' => {
+                self.cursor.advance();
+                single_token!(Category::SlashEqual, start, self.cursor.len_consumed())
+            }
+            '/' => {
+                self.cursor.skip_while(|c| c != '\n');
+                single_token!(Category::Comment, start, self.cursor.len_consumed())
+            }
+            '*' => {
```
Comment (nichtsfrei) [inline rust/nasl-syntax/src/token.rs:381]: change comment parsing
nasl only has:
#1223

File: rust/nasl-syntax/src/token.rs [hunk 3]
```diff
@@ -0,0 +1,773 @@
...
+        )
+    }
+}
+
+// Is used to simplify cases for double_tokens, instead of having to rewrite each match case for each double_token
+// this macro can be used:
+//'+' => double_token!(self.cursor, start, '+', '+', PlusPlus, '=', PlusEqual),
+// within the Iterator implementation of Tokenizer
+macro_rules! double_token {
+    ($cursor:expr, $start:tt, $c:tt, $($l:tt, $bt:expr ), *) => {
```
Comment (nichtsfrei) [inline rust/nasl-syntax/src/token.rs:429]: use better naming

File: rust/nasl-syntax/src/token.rs [hunk 4]
```diff
@@ -0,0 +1,773 @@
...
+            self.cursor.len_consumed()
+        )
+    }
+}
+
+// Is used to simplify cases for double_tokens, instead of having to rewrite each match case for each double_token
+// this macro can be used:
+//'+' => double_token!(self.cursor, start, '+', '+', PlusPlus, '=', PlusEqual),
+// within the Iterator implementation of Tokenizer
+macro_rules! double_token {
```
Comment (nichtsfrei) [inline rust/nasl-syntax/src/token.rs:428]: rename two symbol token

File: rust/nasl-syntax/src/token.rs [hunk 5]
```diff
@@ -0,0 +1,773 @@
...
+        }};
+    }
+
+    #[test]
+    fn skip_white_space() {
+        verify_tokens!("     (       ", vec![(Category::LeftParen, 5, 6)]);
+    }
+
+    #[test]
+    fn single_token() {
```
Comment (nichtsfrei) [inline rust/nasl-syntax/src/token.rs:523]: use symbol naming as within the new macro scheme above

**Review by jjnicola — APPROVED** [review]

**Pull request merged**